sound/soc/sof/card-tplg-ops.c

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2025 Intel Corporation.
+//
+
+#include <linux/device.h>
+#include <linux/errno.h>
+#include "sof-priv.h"
+#include "topology.h"
+
+enum tplg_device_id {
+	TPLG_DEVICE_SDW_JACK,
+	TPLG_DEVICE_SDW_AMP,
+	TPLG_DEVICE_SDW_MIC,
+	TPLG_DEVICE_PCH_DMIC,
+	TPLG_DEVICE_HDMI,
+	TPLG_DEVICE_MAX
+};
+
+#define SDCA_DEVICE_MASK (BIT(TPLG_DEVICE_SDW_JACK) | BIT(TPLG_DEVICE_SDW_AMP) | \
+			  BIT(TPLG_DEVICE_SDW_MIC))
+
+#define PLATFORM_NAME_MAX 4
```

Comment:
I think 4 is enough for Intel, but maybe not long enough for other vendors ?

Reply:
I will rename it to `INTEL_PLATFORM_NAME_MAX` as it will be used by Intel cards only.